tests/Client/TestRepo.php

```diff
@@ -34,31 +29,23 @@ class TestRepo implements RepoFileFetcherInterface
      */
     public function __construct(string $fixturesSet)
     {
-        $this->fixturesSet = $fixturesSet;
+        // Store all the repo files locally so they can be easily altered.
+        // @see self::setRepoFileNestedValue()
+        $repoFiles = glob(static::getFixturesRealPath($fixturesSet, '/tufrepo/metadata') . '/*.json');
+        foreach ($repoFiles as $repoFile) {
+            $this->repoFilesContents[basename($repoFile)] = file_get_contents($repoFile);
```

Comment:
I feel like this is going to potentially chew up a lot of memory. Is this really the right way to go?

Reply:
It will use more memory but this is only for testing. Right this is at most 26 files per test. when I run `composer run test --profile` I don't see any different in memory usage, both say `8.6MiB` from `main`

The fact that we have the contents in memory also means we can really simplify `fetchFile()` and makes it super easy for us to add the new `removeRepoFile()`. 

I would much rather for code simplicity then worry about memory usage in tests. In main and this PR I can run the test suite with minimum `memory_limit = 20M`. So don't think we are close to a problem